functions/npm-diff.js

```diff
@@ -0,0 +1,218 @@
+const pacote = require('pacote');
+const fetch = require('node-fetch');
+const Diff = require('diff');
+const { validate } = require('./utils/validate');
+
+const fetchJson = async (url, options, errorPrefix) => {
+  const response = await fetch(url, options);
+  const json = await response.json();
+  if (!response.ok) {
+    throw new Error(`${errorPrefix}: ${json.message}`);
+  }
+  return json;
+};
+
+const fetchText = async (url, options, errorPrefix) => {
+  const response = await fetch(url, options);
+  const text = await response.text();
+  if (!response.ok) {
+    throw new Error(`${errorPrefix}: ${text}`);
+  }
+  return text;
+};
+
+const getBaseFile = async ({ baseRepoUrl, baseSha }) => {
+  const baseFile = await fetchText(
+    `${baseRepoUrl}/contents/plugins.json?ref=${baseSha}`,
+    {
+      headers: { Accept: 'application/vnd.github.VERSION.raw' },
```

Comment:
`application/vnd.github.VERSION.raw` returns the raw content instead of json response with base64 encoded content.